src/java/org/apache/cassandra/db/compaction/LeveledManifest.java

```diff
@@ -242,11 +242,17 @@ public synchronized CompactionCandidate getCompactionCandidates()
             // we want to calculate score excluding compacting ones
             Set<SSTableReader> sstablesInLevel = Sets.newHashSet(sstables);
             Set<SSTableReader> remaining = Sets.difference(sstablesInLevel, cfs.getTracker().getCompacting());
-            double score = (double) SSTableReader.getTotalBytes(remaining) / (double)maxBytesForLevel(i, maxSSTableSizeInBytes);
+            long remainingBytesForLevel = SSTableReader.getTotalBytes(remaining);
+            long maxBytesForLevel = maxBytesForLevel(i, maxSSTableSizeInBytes);
+            double score = (double) remainingBytesForLevel / (double) maxBytesForLevel;
             logger.trace("Compaction score for level {} is {}", i, score);
 
             if (score > 1.001)
             {
+                // the highest level should not ever exceed its maximum size
+                if (i == generations.levelCount() - 1)
+                    throw new RuntimeException("L" + i + " should not exceed its maximum size (" + maxBytesForLevel + "), but it has " + remainingBytesForLevel + " bytes");
```

Comment:
nit:  add an annotation to indicate `"L" + i` is the max supported level. It helps to reason that why that level cannot compact further.
```suggestion
                    throw new RuntimeException("L" + i + " (maximum supported level) should not exceed its maximum size (" + maxBytesForLevel + "), but it has " + remainingBytesForLevel + " bytes");
Write
```

Reply:
Done.